src/audio/igo_nr/igo_nr.c

```diff
@@ -0,0 +1,523 @@
+			  cd->config.igo_params.nr_mode1_en);
+		comp_info(dev, "  nr_mode3_en          %d",
+			  cd->config.igo_params.nr_mode3_en);
+		comp_info(dev, "  nr_ul_enable         %d",
+			  cd->config.igo_params.nr_ul_enable);
+		comp_info(dev, "  agc_gain             %d",
+			  cd->config.igo_params.agc_gain);
+		comp_info(dev, "  nr_voice_str         %d",
+			  cd->config.igo_params.nr_voice_str);
+		comp_info(dev, "  nr_level             %d",
+			  cd->config.igo_params.nr_level);
+		comp_info(dev, "  nr_mode1_floor       %d",
+			  cd->config.igo_params.nr_mode1_floor);
+		comp_info(dev, "  nr_mode1_od          %d",
+			  cd->config.igo_params.nr_mode1_od);
+		comp_info(dev, "  nr_mode1_pp_param7   %d",
+			  cd->config.igo_params.nr_mode1_pp_param7);
+		comp_info(dev, "  nr_mode1_pp_param8   %d",
+			  cd->config.igo_params.nr_mode1_pp_param8);
+		comp_info(dev, "  nr_mode1_pp_param10  %d",
+			  cd->config.igo_params.nr_mode1_pp_param10);
+		comp_info(dev, "  nr_mode3_floor       %d",
+			  cd->config.igo_params.nr_mode3_floor);
+		comp_info(dev, "  nr_mode1_pp_param53  %d",
+			  cd->config.igo_params.nr_mode1_pp_param53);
+	}
+}
+
+/* copy and process stream data from source to sink buffers */
+static int32_t igo_nr_copy(struct comp_dev *dev)
+{
+	struct comp_copy_limits cl;
+	struct comp_buffer *sourceb;
+	struct comp_buffer *sinkb;
+	struct igo_nr_comp_data *cd = comp_get_drvdata(dev);
+	int32_t src_frames;
+	int32_t sink_frames;
+
+	comp_dbg(dev, "igo_nr_copy()");
+
+	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
+				  sink_list);
+	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
+				source_list);
+
+	/* Check for changed configuration */
+	if (comp_is_new_data_blob_available(cd->model_handler))
+		igo_nr_get_igo_params(dev);
+
+	/* Get source, sink, number of frames etc. to process. */
+	comp_get_copy_limits(sourceb, sinkb, &cl);
+
+	src_frames = audio_stream_get_avail_frames(&sourceb->stream);
+	sink_frames = audio_stream_get_free_frames(&sinkb->stream);
+
+	comp_dbg(dev, "src_frames = %d, sink_frames = %d.", src_frames, sink_frames);
+
+	/* Run the data consume/produce below */
+	src_frames = MIN(src_frames, IGO_FRAME_SIZE - cd->in_wpt);
+	sink_frames = MIN(sink_frames, IGO_FRAME_SIZE - cd->out_rpt);
+
+	comp_dbg(dev, "consumed src_frames = %d, produced sink_frames = %d.",
+		 src_frames, sink_frames);
+
+	igo_nr_process(dev, sourceb, sinkb, src_frames, sink_frames,
+		       src_frames * cl.source_frame_bytes,
+		       sink_frames * cl.sink_frame_bytes);
+
+    /* Run algorithm if buffers are available */
+	if ((cd->in_wpt == IGO_FRAME_SIZE) && (cd->out_rpt == IGO_FRAME_SIZE) {
+		comp_dbg(dev, "algo run");
+
+		if (cd->config.igo_params.nr_bypass) {
+			memcpy_s(cd->out, IGO_FRAME_SIZE * sizeof(int16_t),
+				 cd->in, IGO_FRAME_SIZE * sizeof(int16_t));
+		} else {
+			cd->igo_stream_data_in.data = cd->in;
+			cd->igo_stream_data_in.data_width = IGO_DATA_16BIT;
```

Comment:
These codes are used to configure library's audio data format, not SOF's audio data format. 

For 32-bit audio passing to the component, it will be truncated to 16-bit inside `igo_nr_capture_s32` before sending to `IgoLibProcess`.